### PR TITLE
en_fu OK

### PR DESCRIPTION
**Pull request by fig02**
Fully decompiles the windmill guy actor. I had a hard time naming some functions so I left them unnamed, any suggestions welcome.

### CODE REVIEW

File: src/overlays/actors/ovl_En_Fu/z_en_fu.c [hunk 1]
```diff
@@ -27,33 +37,280 @@ const ActorInit En_Fu_InitVars = {
...
+        this->actionFunc = actionFunc;
+        return 1;
+    }
+    this->actor.textId = textID;
+    yawDiff = this->actor.rotTowardsLinkY - this->actor.shape.rot.y;
+
+    if ((ABS(yawDiff) < 0x2301) && (this->actor.xzDistanceFromLink < 100.0f)) {
+        func_8002F2CC(&this->actor, globalCtx, 100.0f);
+    } else {
+        this->behavior |= FU_RESET_LOOK_ANGLE;
```
[the remark this answers is not in the document]
Comment (fig02) [reply src/overlays/actors/ovl_En_Fu/z_en_fu.c:121]: would a define have the whole bitwise comparison in it? like:
#define FU_RESET_LOOK_ANGLE (this->behaviorFlags & 1)
if so do I need a define for unsetting it?

[the remark this answers is not in the document]
Comment (Roman971) [reply src/overlays/actors/ovl_En_Fu/z_en_fu.c:121]: What I mean by flag defines is just something like `#define FU_RESET_LOOK_ANGLE (1 << 0)` with the other ones being (1 << 1), (1 << 2), etc.

File: src/overlays/actors/ovl_En_Floormas/z_en_floormas.c [hunk 2]
```diff
@@ -52,10 +52,10 @@ const ActorInit En_Floormas_InitVars = {
     FLAGS,
     OBJECT_WALLMASTER,
     sizeof(EnFloormas),
-    EnFloormas_Init,
-    EnFloormas_Destroy,
-    EnFloormas_Update,
-    EnFloormas_Draw,
+    (ActorFunc)EnFloormas_Init,
```
Comment (krimtonz) [inline src/overlays/actors/ovl_En_Floormas/z_en_floormas.c:55]: Are we adding casts to these?

File: src/overlays/actors/ovl_En_Fu/z_en_fu.c [hunk 3]
```diff
@@ -27,33 +39,274 @@ const ActorInit En_Fu_InitVars = {
...
+};
+
+static Vec3f sMtxSrc = {
+    700.0f,
+    700.0f,
+    0.0f,
+};
+
+static UNK_PTR sEyesSegments[] = {
+    0x06005F20,
```
Comment (krimtonz) [inline src/overlays/actors/ovl_En_Fu/z_en_fu.c:56]: Shouldn't these be pointers?
```suggestion
    D_06005F20,
```

File: src/overlays/actors/ovl_En_Fu/z_en_fu.c [hunk 4]
```diff
@@ -27,33 +39,274 @@ const ActorInit En_Fu_InitVars = {
...
+    0x06006320,
+};
+
+static UNK_PTR sMouthSegments[] = {
+    0x06006720,
```
Comment (krimtonz) [inline src/overlays/actors/ovl_En_Fu/z_en_fu.c:61]: same as above.

File: src/overlays/actors/ovl_En_Fu/z_en_fu.c [hunk 5]
```diff
@@ -27,33 +39,274 @@ const ActorInit En_Fu_InitVars = {
...
+    }
+    this->actor.textId = textID;
+    yawDiff = this->actor.rotTowardsLinkY - this->actor.shape.rot.y;
+
+    if ((ABS(yawDiff) < 0x2301) && (this->actor.xzDistanceFromLink < 100.0f)) {
+        func_8002F2CC(&this->actor, globalCtx, 100.0f);
+    } else {
+        this->behaviorFlags |= FU_RESET_LOOK_ANGLE;
+    }
+    return 0;
```
Comment (Random06457) [inline src/overlays/actors/ovl_En_Fu/z_en_fu.c:120]: I believe true and false can be used here
```suggestion
        return true;
    }
    this->actor.textId = textID;
    yawDiff = this->actor.rotTowardsLinkY - this->actor.shape.rot.y;

    if ((ABS(yawDiff) < 0x2301) && (this->actor.xzDistanceFromLink < 100.0f)) {
        func_8002F2CC(&this->actor, globalCtx, 100.0f);
    } else {
        this->behaviorFlags |= FU_RESET_LOOK_ANGLE;
    }
    return false;
```

File: src/overlays/actors/ovl_En_Fu/z_en_fu.c [hunk 6]
```diff
@@ -27,33 +39,274 @@ const ActorInit En_Fu_InitVars = {
...
+};
+
+static UNK_PTR sEyesSegments[] = {
+    0x06005F20,
+    0x06006320,
+};
+
+static UNK_PTR sMouthSegments[] = {
+    0x06006720,
+    0x06006920,
```
Comment (Random06457) [inline src/overlays/actors/ovl_En_Fu/z_en_fu.c:62]:
```suggestion
static UNK_PTR sEyesSegments[] = {
    D_06005F20,
    D_06006320,
};

static UNK_PTR sMouthSegments[] = {
    D_06006720,
    D_06006920,
```

**Pull request merged**